glean-core/src/lib.rs

```diff
@@ -1131,10 +1131,19 @@ pub fn glean_test_destroy_glean(clear_stores: bool, data_path: Option<String>) {
         // and set up the storage.
         let has_storage = core::with_opt_glean(|glean| {
             // We need to flush the ping lifetime data before a full shutdown.
-            glean
-                .storage_opt()
-                .map(|storage| storage.persist_ping_lifetime_data())
-                .is_some()
+            let storage = glean.storage_opt();
+
+            #[cfg(target_os = "android")]
+            {
+                storage
+                    .map(|storage| storage.persist_ping_lifetime_data())
+                    .is_some()
+            }
+
+            #[cfg(not(target_os = "android"))]
+            {
+                storage.is_some()
+            }
```

Comment:
This seems like it's liable to blow up in our faces. Why wouldn't we want to persist ping lifetime data on shutdown in tests? We [do it in real life](https://searchfox.org/glean/source/glean-core/src/lib.rs#690).

Reply:
Mostly to get the test fixed. But yes.
Unfortunately that means I might need to disable the test and potentially move it to one of the instrumentation tests instead.

Reply:
Well, since this is test-only code how about we leave it as you wrote and we annotate it with the bug number of the follow-up for "Make this work in consistently" you're about to file?

Reply:
too late